Week2/JonnathanSanchez/Kata_Rest/spring-rest/src/main/java/springbootcamp/springrest/controllers/ChallengeController.java

```diff
@@ -0,0 +1,90 @@
+package springbootcamp.springrest.controllers;
+
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import springbootcamp.springrest.controllers.domain.FileResult;
+import springbootcamp.springrest.controllers.domain.Student;
+import springbootcamp.springrest.controllers.domain.StudentResult;
+import springbootcamp.springrest.controllers.exception.ChallengeForbiddenException;
+import springbootcamp.springrest.controllers.exception.ChallengePreconditionFailedException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.summarizingInt;
+
+@RestController()
+@RequestMapping("/controller")
+public class ChallengeController {
+
+    private static List<Student> students;
+
+    static {
+        students = new ArrayList<>();
+        students.add(new Student("Jonnathan Sanchez", "1234", 25));
+        students.add(new Student("Jonnathan Delgado", "4567", 23));
+        students.add(new Student("Andrea Medina", "1256", 20));
+        students.add(new Student("Andrea Bedolla", "4597", 25));
+        students.add(new Student("Alejandro Alarcon", "6897", 20));
+        students.add(new Student("Alejandra Perez", "4683", 24));
+        students.add(new Student("Diego Hernandez", "6594", 23));
+        students.add(new Student("Pepe Caceres", "1235", 18));
+    }
+
+    @GetMapping
+    public String getChallenge(){
+        return "Kata_Rest";
+    }
+
+    @GetMapping("/validateDates")
```

Comment:
Those are not pretty good names for resources. For this exercise is fine, however, you should focus more on resources like roles, files, etc.